src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,94 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String COMA = ",";
+    private static final String SPACE = " ";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String fileContent = readFileContent(fromFileName);
+        String[] splitText = splitTextIntoWords(fileContent);
+        int[] results = calculateResult(splitText);
+        String report = createReport(results);
+        writeContentToFile(report, toFileName);
+    }
+
+    private String readFileContent(String fromFileName) {
+        StringBuilder allText = new StringBuilder();
+        File file = new File(fromFileName);
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                allText.append(line).append(SPACE);
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("File not found", e);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file", e);
+        }
+        return allText.toString();
+    }
+
+    private String[] splitTextIntoWords(String text) {
```

Comment:
The 'splitTextIntoWords' method name is misleading because you are not splitting text into words but rather into entries or records. Consider renaming it to 'splitTextIntoRecords' or similar.